src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -47,6 +51,7 @@ internal class GrpcWorker : IWorker
 
             _outputReader = outputChannel.Channel.Reader;
             _outputWriter = outputChannel.Channel.Writer;
+            _hostApplicationLifetime = hostApplicationLifetime;
```

Comment:
Consider adding a null check and throw expresssion.

````
hostApplicationLifetime ?? throw new ArgumentNullException(nameof(hostApplicationLifetime));
````

Reply:
ofc!